internal/container/client.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package container
+
+import (
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/coredata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/general"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+)
+
+var MetadataDeviceClientName = "MetadataDeviceClient"
```

Comment:
All other containers we use `di.TypeInstanceToName` to set the name. This ensures that we have a unique name across all items that could be put into the container. 

```
var MetadataDeviceClientName = di.TypeInstanceToName((*metadata.DeviceClient)(nil))
```

Reply:
updated